### PR TITLE
Migrated boosted trees model to TensorFlow 2.0 in boosted_trees_upgraded folder.

### PR DESCRIPTION
**Pull request by Nimishkhurana**
Preliminary work for GSoC

### CODE REVIEW

**Comment by tfboyd** [comment]
@tanzhenyu will be the main reviewer and will handle this once canned estimators work in TF 2.0.  Target is 15-APR.

**Comment by Nimishkhurana** [comment]
Ok @tfboyd.

File: official/boosted_trees_upgraded/README.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
+# Classifying Higgs boson processes in the HIGGS Data Set
+## Overview
+The [HIGGS Data Set](https://archive.ics.uci.edu/ml/datasets/HIGGS) contains 11 million samples with 28 features, and is for the classification problem to distinguish between a signal process which produces Higgs bosons and a background process which does not.
+
+We use Gradient Boosted Trees algorithm to distinguish the two classes.
+
+---
+
+The code sample uses the high level `tf.estimator.Estimator` and `tf.data.Dataset`.  These APIs are great for fast iteration and quickly adapting models to your own datasets without major code overhauls.  It allows you to move from single-worker training to distributed training, and makes it easy to export model binaries for prediction.  Here, for further simplicity and faster execution, we use a utility function `tf.contrib.estimator.boosted_trees_classifier_train_in_memory`.  This utility function is especially effective when the input is provided as in-memory data sets like numpy arrays.
```
Comment (tanzhenyu) [inline official/boosted_trees_upgraded/README.md]: tf.contrib.xxx no longer exists, use its replacement, i.e., train_in_memory

File: official/boosted_trees_upgraded/README.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
+# Classifying Higgs boson processes in the HIGGS Data Set
```
Comment (tanzhenyu) [inline official/boosted_trees_upgraded/README.md]: High level question (for @tfboyd  as well): why don't we just make sure this model is 1.x and 2.0 compatible? The major changes that I see needed is to have HIGGS dataset as our new tfds. (Assuming 1.x supports tfds)

File: official/boosted_trees_upgraded/data_download.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    raise ValueError("data_dir already has the processed data file: {}".format(
+        np_filename))
+  if not tf.io.gfile.exists(data_dir):
+    tf.io.gfile.mkdir(data_dir)
+  # 2.8 GB to download.
+  try:
+    tf.compat.v1.logging.info("Data downloading...")
+    temp_filename, _ = urllib.request.urlretrieve(input_url)
+    # Reading and parsing 11 million csv lines takes 2~3 minutes.
+    tf.compat.v1.logging.info("Data processing... taking multiple minutes...")
```
Comment (tanzhenyu) [inline official/boosted_trees_upgraded/data_download.py]: could use get_logger. Same for all logging related.

File: official/boosted_trees_upgraded/report.txt [hunk 4]
```diff
@@ -0,0 +1,89 @@
+TensorFlow 2.0 Upgrade Script
```
Comment (tanzhenyu) [inline official/boosted_trees_upgraded/report.txt]: This file does not need to be submitted.

File: official/boosted_trees_upgraded/train_higgs_test.py [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+          train_start=0, train_count=15, eval_start=15, eval_count=5)
+
+    # Loading fine with the correct data_dir.
+    train_data, eval_data = train_higgs.read_higgs_data(
+        self.data_dir,
+        train_start=0, train_count=15, eval_start=15, eval_count=5)
+    self.assertEqual((15, 29), train_data.shape)
+    self.assertEqual((5, 29), eval_data.shape)
+
+ @tf.function
```
Comment (tanzhenyu) [inline official/boosted_trees_upgraded/train_higgs_test.py]: why is this needed?

File: official/boosted_trees_upgraded/README.md (hunk 2 above)
Comment (tfboyd) [reply official/boosted_trees_upgraded/README.md]: One set of code is fine.  We have two sets in situations where we moved from layers to keras or estimator to keras.fit.  If that is not the case here then just upgrade and remove contrib.

File: official/boosted_trees_upgraded/train_higgs_test.py (hunk 5 above)
Comment (Nimishkhurana) [reply official/boosted_trees_upgraded/train_higgs_test.py]: I used tf.function wrapper because original file contains sess.run calls.

File: official/boosted_trees_upgraded/report.txt (hunk 4 above)
Comment (Nimishkhurana) [reply official/boosted_trees_upgraded/report.txt]: ok I will remove it.

File: official/boosted_trees_upgraded/data_download.py (hunk 3 above)
Comment (Nimishkhurana) [reply official/boosted_trees_upgraded/data_download.py]: is get_logger a utility?..I didn't find it. I used absl.logging. Maybe I forgot to replace it in this file. Can I replace it with absl.logging?

File: official/boosted_trees_upgraded/README.md (hunk 1 above)
Comment (Nimishkhurana) [reply official/boosted_trees_upgraded/README.md]: I have used that only, but didn't change it here in readme. I will do it now.

File: official/boosted_trees_upgraded/README.md (hunk 2 above)
Comment (Nimishkhurana) [reply official/boosted_trees_upgraded/README.md]: I also agree with @tfboyd.

**Comment by jaeyounkim** [comment]
I am closing this PR because the boosted_trees directory is deprecated.
